src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1555,21 +1556,21 @@ public bool IsExecutingInvocation(string invocationId)
             return _executingInvocations.ContainsKey(invocationId);
         }
 
-        public bool TryFailExecutions(Exception workerException)
+        public void Shutdown(Exception workerException)
         {
-            if (workerException == null)
-            {
-                return false;
-            }
+            WorkerShutdownException shutdownException = workerException is not null
+                ? new WorkerShutdownException("Worker encountered a fatal error and is shutting down.", workerException)
+                : new WorkerShutdownException("Worker encountered a fatal error and is shutting down.");
 
             foreach (var invocation in _executingInvocations?.Values)
             {
                 string invocationId = invocation.Context?.ExecutionContext?.InvocationId.ToString();
                 _workerChannelLogger.LogDebug("Worker '{workerId}' encountered a fatal error. Failing invocation: '{invocationId}'", _workerId, invocationId);
-                invocation.Context?.ResultSource?.TrySetException(workerException);
+
+                invocation.Context?.ResultSource?.TrySetException(shutdownException);
```

Comment:
Is this the exception which ultimately gets exposed to extensions via the `TryExecuteAsync` and the `FunctionResult.Exception` property?

If so, changing the exception type on them is a breaking change.

Reply:
Discussed offline - writing another comment to explain this and other concerns, posting soon.